src/layer.rs

```diff
@@ -129,18 +131,18 @@ impl LayerSet {
     }
 
     /// Returns an iterator over the names of all layers.
-    pub fn names(&self) -> impl Iterator<Item = &LayerName> {
+    pub fn names(&self) -> impl Iterator<Item = &Name> {
         self.layers.iter().map(|l| &l.name)
     }
 
     /// Returns a new layer with the given name.
-    pub fn new_layer(&mut self, name: &str) -> Result<(), NamingError> {
-        if self.layers.iter().any(|l| &*l.name == name) {
-            Err(NamingError::Duplicate(name.into()))
+    pub fn new_layer(&mut self, name: &str) -> Result<&mut Layer, NamingError> {
```

Comment:
Oops! Did we forget returning something previously?

Reply:
good catch! yes, noticed while updating the signatures that this didn't return anything but probably should.